includes/discovery/sensors/power/eltex-mes23xx.inc.php

```diff
@@ -19,46 +19,45 @@
  * @package    LibreNMS
  * @link       https://www.librenms.org
  *
- * @copyright  2022 Peca Nesovanovic
+ * @copyright  2025 Peca Nesovanovic
  *
  * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
  */
-$divisor = 1000;
-$multiplier = 1;
-if ($pre_cache['eltex-mes23xx-poe']) {
-    foreach ($pre_cache['eltex-mes23xx-poe'] as $index => $data) {
+
+use LibreNMS\Util\Oid;
+
+$oids = SnmpQuery::hideMib()->walk('MARVELL-POE-MIB::rlPethPsePortPowerLimit')->table(2);
+$oids = SnmpQuery::hideMib()->walk('MARVELL-POE-MIB::rlPethPsePortOutputPower')->table(2, $oids);
+
+foreach ($oids as $unit => $indexData) {
+    foreach ($indexData as $ifIndex => $data) {
         if (isset($data['rlPethPsePortOutputPower'])) {
             $value = $data['rlPethPsePortOutputPower'] / $divisor;
             if ($value) {
-                $high_limit = isset($data['rlPethPsePortPowerLimit']) ? ($data['rlPethPsePortPowerLimit'] / $divisor) : null;
-                $high_warn_limit = isset($data['rlPethPsePortPowerLimit']) ? ($data['rlPethPsePortPowerLimit'] / $divisor) * 0.8 : null;
-                $low_warn_limit = 0;
-                $low_limit = 0;
-                [$unit, $ifIndex] = explode('.', $index);
                 $tmp = get_port_by_index_cache($device['device_id'], $ifIndex);
```

Comment:
I just sent a PR changing this to use PortCache

Reply:
ok, done same as you, if yours get merged first, then i will rebase